### PR TITLE
Changes to support auth service URL configuration

### PR DESCRIPTION
**Pull request by kkellerlbl**
Added kb_sdk as a dependency, use kb-sdk to compile spec in Makefile
Added kbase.yml
Added auth url option to deploy.cfg
Added KB_SERVICE_NAME to start_service.tt to let Service.pm read in
same config stanza as Impl
Removed duplicate funcdef in spec file, updated Impl file to match

PR to dev-perl_auth branch, we can merge to develop when ready

### CODE REVIEW

**Comment by kkellerlbl** [comment]
Depends on https://github.com/kbase/kb_sdk/pull/191

File: Makefile [hunk 1]
```diff
@@ -315,17 +318,14 @@ compile-docs: build-libs
...
-	touch lib/biokbase/__init__.py #do not include code in biokbase/__init__.py
-	touch lib/biokbase/$(SERVICE_NAME)/__init__.py
+	kb-sdk compile $(SPEC_FILE) \
+                --out $(LIB_DIR) \
+		--plpsginame $(SERVICE_CAPS).psgi  \
+		--plimplname Bio::KBase::$(SERVICE_CAPS)::$(SERVICE_CAPS)Impl \
+		--plsrvname Bio::KBase::$(SERVICE_CAPS)::Service \
+		--plclname Bio::KBase::$(SERVICE_CAPS)::Client \
+		--pyclname biokbase/$(SERVICE_CAPS)/Client \
+		--jsclname javascript/$(SERVICE_CAPS)/Client
```
Comment (MrCreosote) [inline Makefile]: Should compile in the url as previously

File: kbase.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module-name:
```
Comment (MrCreosote) [inline kbase.yml]: Add a note pointing out this is only used by kb-sdk for compiling the version into the base file, and this is not a registered module

File: Makefile (hunk 1 above)
Comment (kkellerlbl) [inline Makefile]: Added to current PR

File: kbase.yml (hunk 2 above)
Comment (kkellerlbl) [inline kbase.yml]: Added to current PR

**Pull request merged**